demos/python_demos/multi_camera_multi_person_tracking/multi_camera_multi_person_tracking.py

```diff
@@ -0,0 +1,146 @@
+"""
+ Copyright (c) 2019 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import argparse
+import time
+import queue
+from threading import Thread
+import json
+
+import cv2 as cv
+
+from utils.network_wrappers import Detector, VectorCNN
+from mc_tracker.mct import MultiCameraTracker
+from utils.misc import read_py_config
+from utils.video import MulticamCapture
+from utils.visualization import visualize_multicam_detections
+
+
+class frames_thread_body:
+    def __init__(self, capture, max_queue_length=2):
+        self.process = True
+        self.frames_queue = queue.Queue()
+        self.capture = capture
+        self.max_queue_length = max_queue_length
+
+    def __call__(self):
+        while self.process:
+            if self.frames_queue.qsize() > self.max_queue_length:
+                time.sleep(0.1)
+            has_frames, frames = self.capture.get_frames()
+            if not has_frames and self.frames_queue.empty():
+                self.process = False
+                break
+            if has_frames:
+                self.frames_queue.put(frames)
+
+
+def run(params, capture, detector, reid):
+    win_name = 'Multi camera tracking'
+    config = {}
+    if len(params.config):
+        config = read_py_config(params.config)
+
+    tracker = MultiCameraTracker(capture.get_num_sources(), reid, **config)
+
+    thread_body = frames_thread_body(capture, max_queue_length=len(capture.captures) * 2)
+    frames_thread = Thread(target=thread_body)
+    frames_thread.start()
+
+    if len(params.output_video):
+        video_output_size = (1920 // capture.get_num_sources(), 1080)
+        fourcc = cv.VideoWriter_fourcc(*'XVID')
+        output_video = cv.VideoWriter(params.output_video,
+                                      fourcc, 24.0,
+                                      video_output_size)
+    else:
+        output_video = None
+
+    while cv.waitKey(1) != 27 and thread_body.process:
+        start = time.time()
+        try:
+            frames = thread_body.frames_queue.get_nowait()
+        except queue.Empty:
+            frames = None
+
+        if frames is None:
+            continue
+
+        all_detections = detector.get_detections(frames)
+        all_masks = [[] for _ in range(len(all_detections))]
+        for i, detections in enumerate(all_detections):
+            all_detections[i] = [det[0] for det in detections]
+            all_masks[i] = [det[2] for det in detections if len(det) == 3]
```

Comment:
How is it possible to get `len(det)` equal to 3?

Reply:
This part of code id designed to use additional segmentation mask. The 3rd component of detection can be a mask.